tests/capture/input_generators/test_attack_point_iterator.py

```diff
@@ -145,3 +146,113 @@ def test_attack_point_iterator_unrestricted_generator_all_args_len():
     output = list(iter(build_attack_points_iterator(config)))
     assert len(output) == config["bunches"] * config["elements"]
     assert len(output) == len(build_attack_points_iterator(config))
+
+
+def test_attack_point_iterator_repeat():
+    config = {
+        "operation": "repeat",
+        "repetitions": 2,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(iter(build_attack_points_iterator(config)))
+    assert output == list(
+        iter(build_attack_points_iterator(
+            config["configuration"]))) * config["repetitions"]
+
+
+def test_attack_point_iterator_repeat_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 2,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == config["repetitions"] * len(
+        config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_zero_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 0,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == 0
+
+
+def test_attack_point_iterator_repeat_zero():
+    config = {
+        "operation": "repeat",
+        "repetitions": 0,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(build_attack_points_iterator(config))
+    assert output == []
+
+
+def test_attack_point_iterator_repeat_infinite_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": -1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    with pytest.raises(AttributeError):
```

Comment:
Technically not according to the design doc. I am ok with changing that. But

1) where is this behavior documented?

2) `AttributeError` is not what an end-user would expect.

Reply:
What should it return then because we can't return infinite as a length

Reply:
There are two solutions:

1) as the design doc suggests iterate at most `sys.maxsize` iterations.

2) cycle infinitely, but then there needs to be a change in the design doc & the error must be informative and documented.

Reply:
So I have implemented infinite with the cycle function but for the len() function should I create a detailed custom exception?

Reply:
That sounds good. And please document it in the docstring.

Reply:
And also don't forget to suggest a change to the design doc.